src/GitHub.Exports.Reactive/Collections/TrackingCollection.cs

```diff
@@ -0,0 +1,889 @@
+#if !DISABLE_REACTIVEUI
+using ReactiveUI;
+#else
+using System.Windows.Threading;
+using System.Threading;
+#endif
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Diagnostics;
+using System.Linq;
+using System.Reactive;
+using System.Reactive.Concurrency;
+using System.Reactive.Disposables;
+using System.Reactive.Linq;
+
+namespace GitHub.Collections
+{
+    /// <summary>
+    /// TrackingCollection is a specialization of ObservableCollection that gets items from
+    /// an observable sequence and updates its contents in such a way that two updates to
+    /// the same object (as defined by an Equals call) will result in one object on
+    /// the list being updated (as opposed to having two different instances of the object
+    /// added to the list).
+    /// It is always sorted, either via the supplied comparer or using the default comparer
+    /// for T
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
```

Comment:
`RemoveItem` is the only method that's not threadsafe, because it's the only method that doesn't go through the pipeline that's set up in the `Listen` method. This is because I haven't yet thought of a way to encode the "Remove" action into a stream of objects coming in through the observable (an item read from the source observable is either something we haven't seen before, so it's going to be added, or something we've seen before, so it's going to be updated).

`RemoveItem` also triggers events from the ObservableCollection, which need to be on the main thread (because otherwise WPF will seriously complain). Being a subclass of ObservableCollection, this class has the same threading restrictions as that one - actions that change the class have to be on the main thread.

A call to `RemoveItem` is an edge case, so no need to add code just for it.